File: flutter/lib/core/utils/date_formatter.dart
import 'package:smart_expense_manager/core/constants/app_constants.dart';

class DateFormatter {
  static String format(DateTime date) {
    final now = DateTime.now();
    final difference = now.difference(date);

    if (difference.inDays == 0) {
      return AppStrings.today;
    } else if (difference.inDays == 1) {
      return AppStrings.yesterday;
    } else if (difference.inDays < 7) {
      return '${difference.inDays}${AppStrings.daysAgo}';
    } else {
      return '${date.day}/${date.month}/${date.year}';
    }
  }
}

File: android_native/app/src/main/java/com/smartexpense/manager/presentation/utils/DateFormatter.kt
package com.smartexpense.manager.presentation.utils

import com.smartexpense.manager.presentation.constants.AppConstants
import com.smartexpense.manager.presentation.constants.AppStrings
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object DateFormatter {
    fun format(date: Date): String {
        val now = Date()
        val diff = now.time - date.time
        val days = diff / AppConstants.millisecondsPerDay

        return when {
            days == 0L -> AppStrings.TODAY
            days == 1L -> AppStrings.YESTERDAY
            days < 7L -> "${days} days ago"
            else -> SimpleDateFormat(AppConstants.DATE_FORMAT, Locale.getDefault()).format(date)
        }
    }
}
